### PR TITLE
Update usage of json_annotatoin, bump pkg:lints

### PR DESCRIPTION
**Pull request by kevmoo**
Also bump min SDK

### CODE REVIEW

**Comment by kevmoo** [comment]
Feel free to squash/merge and delete the branch if you're happy

**Comment by kevmoo** [comment]
Need to double-check the changelog entries...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
-## 1.2.2-dev
+## 1.3.0
 
+- Added support for `funding` field.
```
Comment (devoncarew) [inline CHANGELOG.md:3]: This did ship in 1.2.1 -

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 - Update SDK requirement to `2.17.0`.
 - Fix the unit tests for Dart 3.0.0.
 
 ## 1.2.1
 
-- Added support for `funding` field.
+- Require Dart SDK >= 2.18.0
```
Comment (devoncarew) [inline CHANGELOG.md:9]: And these two should be in the 1.2.2 / 1.3.2 section ...

**Comment by kevmoo** [comment]
@devoncarew – PTAL

**Pull request merged**